lib/smack/SmackRep.cpp

```diff
@@ -201,22 +311,22 @@ const Expr* SmackRep::b2p(const llvm::Value* v) {
 const Expr* SmackRep::i2b(const llvm::Value* v) {
   return Expr::fn(I2B, expr(v));
 }
-const Expr* SmackRep::b2i(const llvm::Value* v) {
-  return Expr::fn(B2I, expr(v));
+const Expr* SmackRep::b2i(const llvm::Value* v, unsigned size) {
+  return Expr::fn(opName(B2I, {size}), expr(v));
 }
 
 const Expr* SmackRep::lit(const llvm::Value* v) {
   using namespace llvm;
-
+  unsigned wd = 0;
   if (const llvm::ConstantInt* ci = llvm::dyn_cast<const llvm::ConstantInt>(v)) {
-    if (ci->getBitWidth() == 1)
-      return Expr::lit(!ci->isZero());
-
+    wd = ci->getBitWidth();
+    if (wd == 1)
+      return Expr::lit(!ci->isZero(), width);
```

Comment:
The line this replaces constructed a Boogie `bool` type literal. This line constructs an `iXX`. Surely this must be incompatible?